src/localplayer.cpp

```diff
@@ -287,16 +287,9 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 	float player_stepheight = (m_cao == nullptr) ? 0.0f :
 		(touching_ground ? m_cao->getStepHeight() : (0.2f * BS));
 
-	// TODO this is a problematic hack.
-	// Use a better implementation for autojump, or apply a custom stepheight
-	// to all players, as this currently creates unintended special movement
-	// abilities and advantages for Android players on a server.
-#ifdef __ANDROID__
-	if (touching_ground)
-		player_stepheight += (0.6f * BS);
-#endif
-
 	v3f accel_f = v3f(0,0,0);
+	v3f old_position = position;
+	v3f old_speed = m_speed;
```

Comment:
You're right – they are different things, though, since the ones I introduced get changed by the `collisionMoveSimple` directly below. So I've changed the names of my versions to `position_before_move` and `speed_before_move`.